commons/src/main/java/org/infinispan/commons/marshall/ProtoStreamMarshaller.java

```diff
@@ -68,4 +77,56 @@ protected ByteBuffer objectToBuffer(Object o, int estimatedSize) throws IOExcept
    public MediaType mediaType() {
       return MediaType.APPLICATION_PROTOSTREAM;
    }
+
+   /**
+    * A type id mapper that maps old and new type ids back and forth.
+    */
+   public static WrappedMessageTypeMapper getLegacyWrappedMessageTypeMapper() {
```

Comment:
I don't think this belongs here, as it should not be exposed to the client. The legacy mapping only needs to be server-side, so we should limit it's exposure to there.

Reply:
It does not, indeed :)  But it's here just to help the client undo the damage done by the server :)))))